control/controlclient/direct.go

```diff
@@ -945,11 +953,16 @@ func (c *Direct) sendMapRequest(ctx context.Context, isStreaming bool, nu Netmap
 	}
 
 	nodeKey := persist.PublicNodeKey()
+
+	c.mu.Lock()
+	discoKey := c.discoPubKey
+	c.mu.Unlock()
```

Comment:
capture this up above where mu is already locked, rather than doing the lock again